### PR TITLE
Watchlist sync

### PR DESCRIPTION
**Pull request by andrecardoso**
* Added watchlist code from local-storage-module
* Refactored storage to match local-storage-module

### CODE REVIEW

**Comment by andrecardoso** [comment]
Sorry for the large PR, but most of this code should be familiar

**Comment by andrecardoso** [comment]
[(3) Sync local watchlist with messaging service](https://trello.com/c/RbX2jNyB/4142-3-sync-local-watchlist-with-messaging-service)

File: src/storage/messaging/update/update.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    return Promise.resolve();
+  },
+  update(message) {
+    const {filePath, version, token} = message;
+
+    return module.exports.updateWatchlistAndMetadata({
+      filePath, version, token, status: "STALE"
+    })
+    .then(()=>{
+      localMessaging.sendFileUpdate({filePath, status: "STALE", version});
```
Comment (santiagonoguez) [inline src/storage/messaging/update/update.js]: sendFileUpdate returns a Promise, so add return keyword or remove curly braces

File: src/storage/messaging/watch/watch.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+function handleFileWatchResult(message) {
+  const {filePath, version, token} = message;
+
+  console.log(`received version ${version} for ${filePath}`);
+
+  const status = token ? 'STALE' : 'CURRENT';
+
+  return update.updateWatchlistAndMetadata({filePath, version, status, token})
+  .then(()=>{
+    localMessaging.sendFileUpdate({filePath, status, version});
```
Comment (santiagonoguez) [inline src/storage/messaging/watch/watch.js]: same as above

File: src/storage/messaging/watch/watch.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    const metaData = db.fileMetadata.get(filePath) || {filePath};
+    metaData.status = metaData.status || 'UNKNOWN';
+
+    if (metaData.status === 'UNKNOWN') {
+      return module.exports.requestMSUpdate(message, metaData);
+    }
+
+    localMessaging.sendFileUpdate(metaData);
```
Comment (santiagonoguez) [inline src/storage/messaging/watch/watch.js]: this promise could be returned

File: src/storage/messaging/watch/watch.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    return Promise.resolve();
+  },
+  msResult(message) {
+    const {filePath, error, folderData, watchlistLastChanged} = message;
+
+    if (error) {
+      localMessaging.sendFileUpdate({filePath, status: "NOEXIST"});
```
Comment (santiagonoguez) [inline src/storage/messaging/watch/watch.js]: also

**Pull request merged**